src/main/java/org/spongepowered/api/world/Locatable.java

```diff
@@ -22,16 +22,13 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.command.source;
-
-import org.spongepowered.api.command.CommandSource;
-import org.spongepowered.api.world.Location;
-import org.spongepowered.api.world.World;
+package org.spongepowered.api.world;
 
 /**
- * Location sources are {@link CommandSource}s that have a current location.
+ * Location sources are anything that have a current location.
  */
-public interface LocatedSource extends CommandSource {
+@FunctionalInterface
+public interface Locatable {
```

Comment:
@kashike Here's the solution I came up with https://github.com/SpongePowered/SpongeAPI/commit/b7df3bce478666c913934e48863fceb12a4f9b3f